R/fct_db_collect_data.R

```diff
@@ -82,6 +82,15 @@ db_slice_rows <- function(
   dplyr::slice_tail(df, n = 1, by = dplyr::all_of(c(group_vars, slice_vars)))
 }
 
+db_get_table <- function(db_path, db_table = "all_review_data") {
+  stopifnot(is.character(db_path))
+  stopifnot(is.character(db_table))
+  
+  con <- get_db_connection(db_path)
+  dplyr::tbl(con, db_table) |> 
+    dplyr::collect()
```

Comment:
Following up on our discussion last week: I did some quick testing and using a SQL query directly is 2x (and up to 10x) faster. It's still only about milliseconds but maybe it is better to switch to use something like below for functions that we are calling within the application non-interactively.

```
con <- get_db_connection(db_path)
sql <- "SELECT * FROM ?db_table;"
query <- DBI::sqlInterpolate(con, sql, db_table = db_table[1])
DBI::dbGetQuery(con, query)
```

Reply:
Updated